runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -176,10 +182,21 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
                         conflictInfo,
                         tokenResponse.getConflictKey(),
                         AbortCause.NEW_SEQUENCER, // in the future, perhaps define a new AbortCause?
-                        TransactionalContext.getCurrentContext()
-                );
+                        TransactionalContext.getCurrentContext());
             }
         }
 
+        ILogData ld = new LogData(DataType.DATA, object);
+        // Acquiring the buffer is necessary for getSizeEstimate
+        // to return a correct value
+        ld.acquireBuffer();
```

Comment:
It would appear that calculating the size of a log data is a common operation. Provide a static helper in ILogData.